awx/main/analytics/collectors.py

```diff
@@ -217,30 +218,34 @@ def job_instance_counts(since):
     return counts
 
 
-@register('query_info', '1.0')
-def query_info(since, collection_type):
+@register('query_info', '1.0', description=_('Metadata about the analytics collected'))
+def query_info(since, collection_type, **kwargs):
     query_info = {}
     query_info['last_run'] = str(since)
     query_info['current_time'] = str(now())
     query_info['collection_type'] = collection_type
     return query_info
 
 
-# Copies Job Events from db to a .csv to be shipped
-@table_version('events_table.csv', '1.1')
-@table_version('unified_jobs_table.csv', '1.0')
-@table_version('unified_job_template_table.csv', '1.0')
-@table_version('workflow_job_node_table.csv', '1.0')
-@table_version('workflow_job_template_node_table.csv', '1.0')
-def copy_tables(since, full_path, subset=None):
-    def _copy_table(table, query, path):
-        file_path = os.path.join(path, table + '_table.csv')
-        file = open(file_path, 'w', encoding='utf-8')
-        with connection.cursor() as cursor:
-            cursor.copy_expert(query, file)
-            file.close()
-        return file_path
+def _copy_table(table, query, path):
+    file_path = os.path.join(path, table + '_table.csv')
+    file = open(file_path, 'w', encoding='utf-8')
+    with connection.cursor() as cursor:
+        cursor.copy_expert(query, file)
+        file.close()
+    # Ensure we actually dumped data, and not just headers
```

Comment:
Spent 20 minutes staring at psycopg to see if there's a better way to do this. I'd really like to find one.

Reply:
`select count(*)` should be a cheap call.  Looks like `created` field is indexed
(but `modified` looks like not indexed)

Reply:
well, reading 2 lines out of csv is probably much quicker than extra query

Reply:
Reading csv is not necessarily quicker. We are comparing the cost of `[select * query + write-csv + read-csv]` vs `select count(*)` for an empty bundle.  The cost also depends on the probability of an empty 20-minutes window.  I guess the difference won't have much impact to the big picture.

However a `select count(*)` check is much cleaner/clearer in code readability IMHO.

Reply:
In current state maybe, but we want to slowly get to place where we load a bigger chunk from DB and divide the csv (if it goes over 100MB) rather than doing many small queries. That will be more effective on tower side and analytics processor. 

Then this workflow is fine :-)

Reply:
Given how we pass in the query (and mangle it in the unit & integration tests), I'd rather not redo this for this PR. But good to keep in mind if we change this later.

Reply:
Technically speaking @wenottingham, it might actually be better to implement the splitting here _at this layer_ instead of writing one file and then chunking it up into multiple later.  You've already got a file-like interface here; you _could_ count bytes as you go, and swap file references at certain thresholds.

Reply:
@ryanpetrello I think @jameswnl referred to `quick count` if PG would do index scan. So e.g. `select count(*) from events where created > ... and created < ...` should do index scan, if it's small enough portion of the table. (it should be quicker than fetching that timeframe, since it just does scan+count)

But yeah, if PG will choose sequential scan for the the count or the copy on large table, the perf will be horrible. :-)

Reply:
Yeah, if we are uncertain on the cost of `select count(*)`, we should stay with the conservative one.